packages/dashboard-core-plugins/src/controls/markdown/MarkdownEditor.tsx

```diff
@@ -1,15 +1,31 @@
-import React, { PureComponent } from 'react';
-import PropTypes from 'prop-types';
+import React, { PureComponent, ReactElement } from 'react';
 import Markdown from 'react-markdown';
+import { CodeComponent } from 'react-markdown/src/ast-to-react';
 import { Code, Editor } from '@deephaven/console';
+import type * as monaco from 'monaco-editor/esm/vs/editor/editor.api.js';
 
-export default class MarkdownEditor extends PureComponent {
-  constructor(props) {
+interface MarkdownEditorProps {
+  isEditing: boolean;
+  content: string;
```

Comment:
For these, when you have defaultProps set, typescript automatically recognizes that these props are optional when the component is used. So we should keep the types in `MarkdownEditorProps` not optional since inside the component, they are never optional.

Reply:
Good call, my bad